### PR TITLE
Implement stm32f4 inputs/output control

### PR DESCRIPTION
**Pull request by Palantir555**
This PR is work in progress at time of opening.

I'm still debugging the crash caused by the EXTI interrupt associated to the Nucleo's button.

The rest of the code is operational. Run this anywhere in `main.c` (after UART init) to test: 

```
SensorAxes_t sensor_axes;
float sensor_float;
io_nucleoboard_init();
io_sensorboard_init();
io_sensorboard_enable();
while (1)
{
  io_read_gyro(&sensor_axes);
  io_read_accelero(&sensor_axes);
  io_read_magneto(&sensor_axes);
  io_read_pressure(&sensor_float);
  io_read_temperature(&sensor_float);
  io_read_humidity(&sensor_float);
  io_led_off();
  HAL_Delay(250);
  io_led_on();
}
```

### CODE REVIEW

File: examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Inc/mems_cube_hal.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/**
```
Comment (atigyi) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Inc/mems_cube_hal.h]: are demo_io.h and mems_cube_hal.h files from the mems SDK? Are they intact? If we use the "stock" version than we shouldn't put it into our sources but link it from the mems SDK. Probably we already have to have a variable containing the path to mems SDK, don't we?
I know that Milan compied all the files into current wifi examples. I just removed it in recent stories and referencing to files in the Wifi SDK.

File: examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c [hunk 2]
```diff
@@ -0,0 +1,358 @@
+#include <assert.h>
```
Comment (atigyi) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c]: the same here if this demo_io.c is not modified, then use it from the mems SDK as a link.

Comment (atigyi) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c]: ok, downloaded the mems SDK and haven't found this file, so you put this file together?

Comment (Palantir555) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c]: Nope, I didn't 🙃  I'll implement a custom error handler to loop forever on error  and get rid of the `assert()`s

Comment (Palantir555) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c]: Sorry, I misunderstood your comments. Yeah, this file is the interface I wrote so we can keep all GPIO code away from `main.c`

Comment (atigyi) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/demo_io.c]: ok, this is great :+1:

**Comment by atigyi** [comment]
Great job, approved!!!

File: examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/stm32_xx_it.c [hunk 3]
```diff
@@ -69,6 +70,38 @@ void TIMp_IRQHandler(void);
 /* Private functions ---------------------------------------------------------*/
 
 /******************************************************************************/
+/*                Demo GPIO Interrupt implementation                         */
+/******************************************************************************/
```
Comment (Palantir555) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/stm32_xx_it.c]: @atigyi I just pushed the fix for the button interrupt. This is the biggest change, the rest of the modifications were for style or clean-up. If these 2 functions look good to you, I'll merge the branch :)

File: examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Inc/mems_cube_hal.h (hunk 1 above)
Comment (Palantir555) [inline examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Inc/mems_cube_hal.h]: You're right. I thought I still had something necessary in that file, but I didn't. I've removed it.

**Pull request merged**